### PR TITLE
Add view translate_join, index now stores original sequences

### PR DESCRIPTION
**Pull request by sarahet**
* Index now stores the original sequences and not the translated sequences
* Index does not store sequence lengths anymore, they get calculated from the original sequences when needed
* Added view_translate_join
* Fix setting of thread number for search and index
* Updated submodules

### CODE REVIEW

File: src/search_output.hpp [hunk 1]
```diff
@@ -347,18 +347,25 @@ myWriteHeader(TGH & globalHolder, TLambdaOptions const & options)
...
+        //     SEQAN_OMP_PRAGMA(parallel for simd)
+        //     for (size_t i = 0; i < globalHolder.redSubjSeqs.size(); ++i)
+        //         subjSeqLengths[i] = globalHolder.indexFile.seqs[i].size();
+        // }
+
+        // compute seqan::lengths ultra-fast
+        seqan::resize(subjSeqLengths, globalHolder.redSubjSeqs.size());
+        SEQAN_OMP_PRAGMA(parallel for simd)
+        for (size_t i = 0; i < globalHolder.redSubjSeqs.size(); ++i)
+            subjSeqLengths[i] = globalHolder.indexFile.seqs[i].size();
```
Comment (h-2) [inline src/search_output.hpp]: why are these computed at all?

File: src/seqan2seqan3.hpp [hunk 2]
```diff
@@ -35,10 +36,20 @@ template <typename t, auto j>
...
+template <typename ...ts>
+inline constexpr bool is_new_range<seqan3::detail::view_translate_single<ts...>> = true;
+template <typename t, bool b1, bool b2>
+inline constexpr bool is_new_range<seqan3::detail::view_take<t, b1, b2>> = true;
 
 template <typename t>
 SEQAN3_CONCEPT NonSeqAn2Range = is_new_range<seqan3::remove_cvref_t<t>>;
 
+static_assert(NonSeqAn2Range<seqan3::detail::view_take<ranges::drop_view<seqan3::detail::view_translate_single<ranges::ref_view<std::vector<seqan3::dna5> > > >, false, false>>, "not a nonseqan2range");
+
```
Comment (h-2) [inline src/seqan2seqan3.hpp]: This can be removed

File: src/shared_definitions.hpp [hunk 3]
```diff
@@ -194,3 +198,59 @@ struct fake_index_file
...
+struct lazy_conditional<true, on_true_t, on_false_t>
+{
+    using type = instantiate_t<on_true_t>;
+};
+
+template <typename on_true_t, typename on_false_t>
+struct lazy_conditional<false, on_true_t, on_false_t>
+{
+    using type = instantiate_t<on_false_t>;
+};
```
Comment (h-2) [inline src/shared_definitions.hpp]: remove lazy stuff in favour of seqan3

File: src/search_algo.hpp [hunk 4]
```diff
@@ -284,13 +290,20 @@ loadDbIndexFromDisk(TGlobalHolder       & globalHolder,
     if constexpr (seqan::sIsTranslated(TGlobalHolder::blastProgram))
     {
         // last value has sum of lengths
-        seqan::context(globalHolder.outfileBlastTab).dbTotalLength  = globalHolder.indexFile.origSeqLengths.back();
-        seqan::context(globalHolder.outfileBlastTab).dbNumberOfSeqs = globalHolder.indexFile.origSeqLengths.size() - 1;
+        // seqan::context(globalHolder.outfileBlastTab).dbTotalLength  = globalHolder.indexFile.origSeqLengths.back();
+        // seqan::context(globalHolder.outfileBlastTab).dbNumberOfSeqs = globalHolder.indexFile.origSeqLengths.size() - 1;
+        seqan::context(globalHolder.outfileBlastTab).dbTotalLength  =
+            std::accumulate(globalHolder.indexFile.seqs.begin(), globalHolder.indexFile.seqs.end(), 0,
```
Comment (h-2) [inline src/search_algo.hpp]:
```suggestion
            std::ranges::accumulate(globalHolder.indexFile.seqs, 0,
```

File: src/search_algo.hpp [hunk 5]
```diff
@@ -284,13 +290,20 @@ loadDbIndexFromDisk(TGlobalHolder       & globalHolder,
...
+                [](int sum, std::vector<typename TGlobalHolder::TOrigSbjAlph> a)
```
Comment (h-2) [inline src/search_algo.hpp]:
```suggestion
                [](size_t sum, std::vector<typename TGlobalHolder::TOrigSbjAlph> const & a)
```

Please do not copy every sequence when adding their sizes :scream:

File: src/search_output.hpp [hunk 6]
```diff
@@ -347,18 +347,25 @@ myWriteHeader(TGH & globalHolder, TLambdaOptions const & options)
...
+        //     seqan::copy_range(globalHolder.indexFile.origSeqLengths, subjSeqLengths);
+        //     seqan::resize(subjSeqLengths, seqan::length(subjSeqLengths) - 1);
+        // } else
+        // {
+        //     // compute seqan::lengths ultra-fast
+        //     seqan::resize(subjSeqLengths, globalHolder.redSbjSeqs.size());
+        //     SEQAN_OMP_PRAGMA(parallel for simd)
+        //     for (size_t i = 0; i < globalHolder.redSbjSeqs.size(); ++i)
+        //         subjSeqLengths[i] = globalHolder.indexFile.seqs[i].size();
+        // }
```
Comment (h-2) [inline src/search_output.hpp]: remove commented out code

File: src/shared_definitions.hpp [hunk 7]
```diff
@@ -148,16 +151,18 @@ template <typename TString>
 using TCDStringSet = std::vector<TString>; //TODO seqan3::concatenated_sequences
 
 
+// template <DbIndexType           dbIndexType,
+//           AlphabetEnum          origAlph,
+//           AlphabetEnum          transAlph>    // <- all members of index_file_options that influence types
+
```
Comment (h-2) [inline src/shared_definitions.hpp]: remove

**Comment by h-2** [comment]
build fails. seems like we have not aliased ranges::accumulate to std::ranges accumulate. Then stick with the regular one for now.

**Comment by sarahet** [comment]
Now, only gcc9 fails. Was that expected?

**Comment by h-2** [comment]
That's definitely unrelated (although we should fix mac support at some point).

**Pull request merged**